src/wp-includes/rest-api/endpoints/class-wp-rest-widget-types-controller.php

```diff
@@ -397,6 +420,71 @@ public function get_item_schema() {
 		return $this->add_additional_fields_schema( $this->schema );
 	}
 
+	public function render( $request ) {
+		return array(
+			'preview' => $this->render_legacy_widget_preview_iframe(
+				$request['id_base'],
+				isset( $request['instance'] ) ? $request['instance'] : null
+			),
+		);
+	}
+
+	/**
+	 * Renders a page containing a preview of the requested Legacy Widget block.
+	 *
+	 * @param string $id_base The id base of the requested widget.
+	 * @param array $instance The widget instance attributes.
+	 *
+	 * @return string Rendered Legacy Widget block preview.
+	 */
+	private function render_legacy_widget_preview_iframe( $id_base, $instance ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
```

Comment:
Why is the permission check here instead of in the `permission_callback`? This should go into its own `permission_callback` specific for this endpoint, that can then call `check_read_permission`.

Reply:
We should actually probably just reuse `get_item_permission_callback` which matches what we do in the encode endpoint.